resilience4j-rxjava2/src/main/java/io/github/resilience4j/bulkhead/operator/DisposableBulkhead.java

```diff
@@ -25,22 +24,53 @@
 
     @Override
     public void dispose() {
-        if (disposed.compareAndSet(false, true)) {
+        if (DisposableHelper.dispose(this)) {
             releaseBulkhead();
-            disposable.dispose();
         }
     }
 
     @Override
     public boolean isDisposed() {
-        return disposed.get();
+        return DisposableHelper.isDisposed(get());
     }
 
-    protected void setDisposable(Disposable disposable) {
-        this.disposable = requireNonNull(disposable);
+    protected void onSubscribe(Disposable disposable, Consumer<Disposable> onSubscribe, Consumer<Throwable> onError) {
```

Comment:
I would have implemented abstract methods instead so no allocation/lambdas have to happen.

Reply:
I refactored that part a bit avoiding lambdas. Thanks.